code/mission/missionmessage.cpp

```diff
@@ -2045,8 +2086,32 @@ bool filters_match(MessageFilter& filter, ship* it) {
 		    && filter_matches(hud_get_ship_class(it), filter.class_name)
 		    && filter_matches(wing_name, filter.wing_name)
 		    && filter_matches(Ship_info[it->ship_info_index].species, filter.species_bitfield)
-		    && filter_matches(Ship_info[it->ship_info_index].class_type, filter.type_bitfield);
+		    && filter_matches(Ship_info[it->ship_info_index].class_type, filter.type_bitfield)
+		    && filter_matches(it->team, filter.team_bitfield);
+	}
+}
+
+bool outer_filters_match(MessageFilter& filter, int range, ship* sender) {
+	for (auto i: list_range(&Ship_obj_list)) {
+		auto obj = &Objects[i->objnum];
+		// Ignore dying ships
+		if (obj->flags[Object::Object_Flags::Should_be_dead]) {
```

Comment:
Bit of a nitpick here, but `Object::Object_Flags::Should_be_dead` is distinct from `Object::Object_Flags::Dying` and the comment seems to imply that only dying ships are skipped. I think the intent here is possibly to skip checks for both Dead and Dying ships, in which case `Object::Object_Flags::Dying` should also be added here and the comment updated to reflect both scenarios.

Reply:
Done.